File: trunk/Java/OPSJLib/src/ops/McSendDataHandler.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package ops;

import java.io.IOException;
import java.net.InetAddress;
import java.util.Vector;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author Anton Gravestam
 */
public class McSendDataHandler extends SendDataHandlerBase
{
    private final InetAddress sinkIP;

    public McSendDataHandler(Topic t, String localInterface) throws IOException
    {
        sender = new MulticastSender(0, localInterface, 1, t.getOutSocketBufferSize());
        sinkIP = InetAddress.getByName(t.getDomainAddress());
    }

    public synchronized boolean sendData(byte[] bytes, int size, Topic t)
    {
        return sendData(bytes, size, sinkIP, t.getPort());
    }

}

File: trunk/C#/Ops/McSendDataHandler.cs
///////////////////////////////////////////////////////////
//  McSendDataHandler.cs
//  Implementation of the Class McSendDataHandler
//  Created on:      12-nov-2011 09:25:31
//  Author:
///////////////////////////////////////////////////////////

using System.Collections.Generic;
using System.Runtime.CompilerServices;  // Needed for the "MethodImpl" synchronization attribute

namespace Ops 
{
	public class McSendDataHandler : SendDataHandler 
    {
        private readonly InetAddress sinkIP;

        public McSendDataHandler(Topic t, string localInterface) 
        {
            sender = new MulticastSender(0, localInterface, 1, t.GetOutSocketBufferSize());     // Make ttl configurable
            sinkIP = InetAddress.GetByName(t.GetDomainAddress());
        }

        [MethodImpl(MethodImplOptions.Synchronized)]
        public override bool SendData(byte[] bytes, int size, Topic t)
        {
            return SendData(bytes, size, sinkIP, t.GetPort());
        }

	}

}